### DISCUSSION TITLE
q-input with date picker using popup proxy not hiding

### DISCUSSION THREAD

**Discussion by miguelrrsantos** [quasarframework/quasar#15565] [General - Components / Directives / etc]
Hello.

I was trying to make a date picker pop up with popup proxy and when the user clicks on the desired date the popup should close. I implemented the code with composition api and using ref.

However it seems the ref is always null and not capable to target the popup proxy. Here is the code that i implemented:

```
<q-input v-model="proxyPost.publicationDate" debounce="300" :placeholder="`${$t('posts.publication_date')}`" outlined dense clearable mask="date" @blur="proxyVuelidate.post.publicationDate.$touch;" :error="proxyVuelidate.post.publicationDate.$error">
  <template v-slot:append>
    <q-icon name="event" class="cursor-pointer">
      <q-popup-proxy ref="publicationDateComponent" cover transition-show="scale" transition-hide="scale">
        <q-date v-model="proxyPost.publicationDate" @update:model-value="publicationDateComponent.hide()"></q-date>
      </q-popup-proxy>
    </q-icon>
  </template>
  <template v-slot:error>
    <span v-for="$error in proxyVuelidate.post.publicationDate.$errors" :key="$error.$property">{{ `${$error.$message}. ` }}
    </span>
  </template>
</q-input>
```

compostion api, setup code:
```
const publicationDateComponent = ref(null)

return {
            publicationDateComponent,
        }
```

In browser console, i get this error:
`$setup.publicationDateComponent.hide is not a function`

It's strange because i made an exmple that i think it's similar on this codepen and it's working:
[https://codepen.io/miguelrrsantos/pen/dyqeppK](https://codepen.io/miguelrrsantos/pen/dyqeppK)

So, can you give me some explanation of what is the problem? It seems that the popup proxy is hidded and the ref never gets the popup element on mount event.

Best regards,
Miguel.

**Comment by metalsadman** [comment]
if it's working in your codepen but not in your code, then you have some bug somewhere else.
